packages/headless/src/controllers/product-listing/results-per-page/headless-product-listing-results-per-page.ts

```diff
@@ -0,0 +1,65 @@
+import {configuration, pagination} from '../../../app/reducers';
+import {logPagerResize} from '../../../features/pagination/pagination-analytics-actions';
+import {fetchProductListing} from '../../../features/product-listing/product-listing-actions';
+import {ProductListingEngine} from '../../../product-listing.index';
+import {
+  ConfigurationSection,
+  PaginationSection,
+} from '../../../state/state-sections';
+import {loadReducerError} from '../../../utils/errors';
+import {
+  buildCoreResultsPerPage,
+  ResultsPerPage,
+  ResultsPerPageInitialState,
+  ResultsPerPageProps,
+  ResultsPerPageState,
+} from '../../core/results-per-page/headless-core-results-per-page';
+
+export {
+  ResultsPerPage,
+  ResultsPerPageProps,
+  ResultsPerPageInitialState,
+  ResultsPerPageState,
+};
+
+/**
+ * Creates a `ResultsPerPage` controller instance.
+ *
+ * @param engine - The headless engine.
+ * @param props - The configurable `ResultsPerPage` properties.
+ * @returns A `ResultsPerPage` controller instance.
+ */
+export function buildResultsPerPage(
+  engine: ProductListingEngine,
+  props: ResultsPerPageProps = {}
+): ResultsPerPage {
+  if (!loadResultsPerPageReducers(engine)) {
+    throw loadReducerError;
+  }
+
+  const coreController = buildCoreResultsPerPage(engine, props);
+  const {dispatch} = engine;
+
+  return {
+    ...coreController,
+
+    get state() {
+      return {
+        ...coreController.state,
+      };
+    },
+
+    set(num: number) {
+      coreController.set(num);
+      dispatch(fetchProductListing());
+      dispatch(logPagerResize());
```

Comment:
```suggestion
      dispatch(fetchProductListing()).then(() => dispatch(logPagerResize()));
```

I'll be changing this project-wide

Reply:
With product-listing, aren't all analytics managed server-side?

Reply:
Yes and no :( 

The Search API is ready to log back-end calls for the *results*, but not for the component actions such as "changing the page" or "clicking on a facet result". 

So this is an in-between that we'll have to deal with for some time, unfortunately.